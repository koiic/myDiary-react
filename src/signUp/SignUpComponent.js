import React, { Component, Fragment } from 'react';

// third party library
import { Redirect } from 'react-router-dom';
import Loader from 'react-loader';


/**
 * @param {object} props
 * @desc renders signup page
 */
class SignUp extends Component {
  state = {
    firstname: '',
    lastname: '',
    email: '',
    password: '',
    username: '',
  };

  handleChange = (event) => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const {
      username,
      password,
      firstname,
      lastname,
      email
    } = this.state;
    this.props.userRegistration({
      username, password, firstname, lastname, email
    });
  };

  render() {
    const {
      firstname, lastname, email, password, username
    } = this.state;
    const { props } = this;
    const loading = props.auth.signup.processing
      ? { display: 'block' }
      : { display: 'none' };
    if (props.auth.isAuth === true) {
      return <Redirect to='/' />;
    }
    return (
      <Fragment>
        <section>
        <div className="shelter">
        <div className="row">
          <div className="column" />
          <div className="column">
            <div className="form">
              <h1 className="text-center nav__logo app-color">
                Registration
              </h1>
              <div style={loading}>
                        <Loader color='#0FC86F' speed={1} className='spinner' />
                      </div>
              <div className="form-card">
                <form onSubmit={this.handleSubmit} >
                  <div className="form-row">
                    <label htmlFor="name">Firstname:</label>
                    <input
                      type="text"
                      className="form-control"
                      placeholder="Enter your firstname"
                      name="firstname"
                      value={firstname}
                      onChange={this.handleChange}
                      required
                      minLength={3}
                      maxLength={30}
                    />
                  </div>
                  <div className="form-row">
                    <label htmlFor="name">Lastname:</label>
                    <input
                      type="text"
                      className="form-control"
                      placeholder="Enter your lastname"
                      required
                      name="lastname"
                      value={lastname}
                      onChange={this.handleChange}
                      minLength={3}
                      maxLength={30}
                    />
                  </div>
                  <div className="form-row">
                    <label htmlFor="name">Username:</label>
                    <input
                      type="text"
                      className="form-control"
                      placeholder="Enter your username"
                      required
                      name="username"
                      value={username}
                      onChange={this.handleChange}
                      minLength={3}
                      maxLength={30}
                    />
                  </div>
                  <div className="form-row">
                    <label htmlFor="name">Email:</label>
                    <input
                      type="email"
                      className="form-control"
                      placeholder="Enter your email"
                      required
                      name="email"
                      value={email}
                      onChange={this.handleChange}
                      minLength={3}
                      maxLength={30}
                    />
                  </div>
                  <div className="form-row">
                    <label htmlFor="name">Password</label>
                    <input
                      type="password"
                      className="form-control"
                      required
                      name="password"
                      value={password}
                      onChange={this.handleChange}
                      placeholder="Enter your password"
                    />
                  </div>
                    <button
                      className="btn account__btn btn-block"
                    >
                      Sign up
                    </button>
                </form>
              </div>
              <p>
                Already Have an Account Sign In{' '}
                <a href="/login">here</a>
              </p>
            </div>
          </div>
          <div className="column" />
        </div>
      </div>
    </section>
    <section>
      <div className="footer">
        <div className="navbar py-1 bg-white">
          <div className="shelter">

            <nav className="nav__items nav__items-right">
              <div className="navbar__nav">
                <div className="display">
                  <a href="#" className="dodgerblue">
                    Â© Ismail Ibrahim [Andela Cycle34 2018].
                  </a>
                </div>
              </div>
            </nav>
          </div>
          {/* end shelter */}
        </div>
        {/* end navbar */}
      </div>
    </section>
  </Fragment>

    );
  }
}

export default SignUp;
