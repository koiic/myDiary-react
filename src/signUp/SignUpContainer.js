import { connect } from 'react-redux';

// components
import SignUpComponent from './SignUpComponent';

// action
import { userRegistration } from './signUpAction';

const mapDispatchToProps = dispatch => ({
  userRegistration: detail => dispatch(userRegistration(detail))
});
const mapStateToProps = state => ({
  auth: state.authReducer
});

export default connect(
  mapStateToProps, mapDispatchToProps
)(SignUpComponent);
