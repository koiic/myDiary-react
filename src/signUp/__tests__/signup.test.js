// React Libraries
import React from 'react';

// third-party libraries
import { shallow, configure } from 'enzyme';

import { Provider } from 'react-redux';

// modules imports
import configureStore from '../../app/mainStore/store';

// modules importation
import SignUpContainer from '../SignUpContainer';
import SignUpComponent from '../SignUpComponent';

// SnapShot Test
describe('Renders the singup container', () => {
  it('should render the signup form', () => {
    const store = configureStore();
    const wrapper = shallow(
      <Provider store={store}>
        <SignUpContainer />
      </Provider>
    );
    expect(wrapper.exists()).toBe(true);
  });
});

describe('Renders the singup page', () => {
  it('should render the signup form', () => {
    const component = shallow(
      <SignUpComponent auth={{ signup: { processing: true } }}/>
    );
    expect(component).toMatchSnapshot();
  });
});
