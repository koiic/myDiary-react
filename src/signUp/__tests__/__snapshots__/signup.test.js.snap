// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the singup page should render the signup form 1`] = `
<Fragment>
  <section>
    <div
      className="shelter"
    >
      <div
        className="row"
      >
        <div
          className="column"
        />
        <div
          className="column"
        >
          <div
            className="form"
          >
            <h1
              className="text-center nav__logo app-color"
            >
              Registration
            </h1>
            <div
              style={
                Object {
                  "display": "block",
                }
              }
            >
              <Component
                className="spinner"
                color="#0FC86F"
                component="div"
                loadedClassName="loadedContent"
                parentClassName="loader"
                speed={1}
              />
            </div>
            <div
              className="form-card"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-row"
                >
                  <label
                    htmlFor="name"
                  >
                    Firstname:
                  </label>
                  <input
                    className="form-control"
                    maxLength={30}
                    minLength={3}
                    name="firstname"
                    onChange={[Function]}
                    placeholder="Enter your firstname"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-row"
                >
                  <label
                    htmlFor="name"
                  >
                    Lastname:
                  </label>
                  <input
                    className="form-control"
                    maxLength={30}
                    minLength={3}
                    name="lastname"
                    onChange={[Function]}
                    placeholder="Enter your lastname"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-row"
                >
                  <label
                    htmlFor="name"
                  >
                    Username:
                  </label>
                  <input
                    className="form-control"
                    maxLength={30}
                    minLength={3}
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter your username"
                    required={true}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-row"
                >
                  <label
                    htmlFor="name"
                  >
                    Email:
                  </label>
                  <input
                    className="form-control"
                    maxLength={30}
                    minLength={3}
                    name="email"
                    onChange={[Function]}
                    placeholder="Enter your email"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-row"
                >
                  <label
                    htmlFor="name"
                  >
                    Password
                  </label>
                  <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="Enter your password"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn account__btn btn-block"
                >
                  Sign up
                </button>
              </form>
            </div>
            <p>
              Already Have an Account Sign In
               
              <a
                href="/login"
              >
                here
              </a>
            </p>
          </div>
        </div>
        <div
          className="column"
        />
      </div>
    </div>
  </section>
  <section>
    <div
      className="footer"
    >
      <div
        className="navbar py-1 bg-white"
      >
        <div
          className="shelter"
        >
          <nav
            className="nav__items nav__items-right"
          >
            <div
              className="navbar__nav"
            >
              <div
                className="display"
              >
                <a
                  className="dodgerblue"
                  href="#"
                >
                  Â© Ismail Ibrahim [Andela Cycle34 2018].
                </a>
              </div>
            </div>
          </nav>
        </div>
      </div>
    </div>
  </section>
</Fragment>
`;
